rclcpp_lifecycle/test/test_transition_wrapper.cpp

```diff
@@ -125,4 +132,79 @@ TEST_F(TestTransitionWrapper, exceptions) {
   EXPECT_THROW(a->goal_state(), std::runtime_error);
   EXPECT_THROW(a->id(), std::runtime_error);
   EXPECT_THROW(a->label(), std::runtime_error);
+
+  {
+    auto mock = mocking_utils::patch_and_return(
+      "lib:rclcpp_lifecycle", rcl_lifecycle_transition_init, RCL_RET_ERROR);
+
+    EXPECT_THROW(
+      std::make_shared<TransitionDerived>(1, "one"),
+      std::runtime_error);
+
+    rclcpp_lifecycle::State state1(1, "start_state");
+    rclcpp_lifecycle::State state2(2, "goal_state");
+    EXPECT_THROW(
+      std::make_shared<TransitionDerived>(2, "two", std::move(start_state), std::move(goal_state)),
+      std::runtime_error);
+  }
+  {
+    auto mock = mocking_utils::patch_and_return(
+      "lib:rclcpp_lifecycle", rcl_lifecycle_transition_fini, RCL_RET_ERROR);
+    auto transition1 = std::make_shared<TransitionDerived>(1, "one");
+    EXPECT_NO_THROW(transition1->expose_reset());
+
+    rclcpp_lifecycle::State state1(1, "start_state");
+    rclcpp_lifecycle::State state2(2, "goal_state");
+    auto transition2 =
+      std::make_shared<TransitionDerived>(2, "two", std::move(start_state), std::move(goal_state));
+    EXPECT_NO_THROW(transition2->expose_reset());
```

Comment:
Similar question here, this looks as expected behavior without mocking, anyway to check if failed?

Reply:
The error will be logged that memory leaks here, but there is unfortunately no way to check for failure.